gammapy/utils/coordinates/fov.py

```diff
@@ -1,122 +1,69 @@
-
-    # compute new lon, lat
-    lon_t = -np.rad2deg(np.arctan2(y_t, x_t))
-    lat_t = np.rad2deg(np.arcsin(z_t))
-
-    # shift lon by 360 degrees if negative
-    lon_t = np.where(lon_t < 0, lon_t + 360, lon_t)
-
-    return lon_t, lat_t
+
+    # Create a frame that is centered on the pointing position
+    center = SkyCoord(lon_pnt, lat_pnt)
+    fov_frame = SkyOffsetFrame(origin=center)
+
+    # Define coordinate to be transformed.
+    # Need to switch the sign of the longitude angle here
+    # because this axis is reversed in our definition of the FoV-system
+    target_fov = SkyCoord(-lon, lat, frame=fov_frame)
+
+    # Transform into celestial system (need not be ICRS)
+    target_sky = target_fov.icrs
+
+    return target_sky.ra, target_sky.dec
 
 
 def sky_to_fov(lon, lat, lon_pnt, lat_pnt):
     """Make a transformation from sky coordinates to field-of-view coordinates.
 
     Parameters
     ----------
-    lon : array_like
-        Sky longitude coordinate to be transformed, in degrees
-    lat : array_like
-        Sky latitude coordinate to be transformed, in degrees
-    lon_pnt : array_like
-        Longitude coordinate of the pointing position, in degrees
-    lat_pnt : array_like
-        Latitude coordinate of the pointing position, in degrees
+    lon, lat : `~astropy.units.Quantity`
+        Sky coordinate to be transformed
+    lon_pnt, lat_pnt : `~astropy.units.Quantity`
+        Coordinate specifying the pointing position
+        (i.e. the center of the field of view)
 
     Returns
     -------
-    lon_t : array_like
-        Field-of-view longitude coordinate, in degrees
-    lat_t : array_like
-        Field-of-view latitude coordinate, in degrees
+    lon_t, lat_t : `~astropy.units.Quantity`
+        Transformed field-of-view coordinate
     """
-    lon_ = -lon
-
-    # compute cartesian coordinates
-    x = np.cos(np.deg2rad(lat)) * np.cos(np.deg2rad(lon_))
-    y = np.cos(np.deg2rad(lat)) * np.sin(np.deg2rad(lon_))
-    z = np.sin(np.deg2rad(lat))
-
-    # make sure vector is properly normalised
-    assert np.allclose(x ** 2 + y ** 2 + z ** 2, 1)
-
-    # transform
-    lat_pnt = np.deg2rad(90 - lat_pnt)
-    lon_pnt = -np.deg2rad(lon_pnt)
-    x_t = (
-        x * np.cos(lat_pnt) * np.cos(lon_pnt)
-        + y * np.sin(lon_pnt) * np.cos(lat_pnt)
-        - z * np.sin(lat_pnt)
-    )
-    y_t = -x * np.sin(lon_pnt) + y * np.cos(lon_pnt)
-    z_t = (
-        x * np.sin(lat_pnt) * np.cos(lon_pnt)
-        + y * np.sin(lat_pnt) * np.sin(lon_pnt)
-        + z * np.cos(lat_pnt)
-    )
-
-    # switch coordinates due to axis convention
-    x_ = z_t
-    y_ = y_t
-    z_ = -x_t
-
-    # compute new lon, lat
-    lon_t = np.rad2deg(np.arctan2(y_, x_))
-    lat_t = np.rad2deg(np.arcsin(z_))
-
-    return lon_t, lat_t
+
+    # Create a frame that is centered on the pointing position
+    center = SkyCoord(lon_pnt, lat_pnt)
+    fov_frame = SkyOffsetFrame(origin=center)
+
```

Comment:
Yes, this might be simpler, but I did not do it because I wasn't sure what would happen when the user passes a `SkyCoord` that is defined in another system. You're right that currently, the transformation works for the conversion between AltAz and an AltAz-aligned field-of-view system, and not e.g. between RaDec and a RaDec-aligned field-of-view system. The pointing therefore needs to be specified in AltAz coordinates, yes.

My understanding was that Gammapy will work with AltAz-aligned field-of-view systems only, and so I avoided the complication to also support a RaDec-aligned system.

It might be a good idea to make it more clear in the comments that the field-of-view system is always assumed to be AltAz-aligned.